### PR TITLE
Add name/value pair generic file collector.

### PR DESCRIPTION
**Pull request by stantona**
The intention of this collector is to gather metrics found
in files. Candidates could be standard proc files such as meminfo or 
vmstat but could be used for any file that follows a name/value format.

Given a regex and a file with metrics, the collector can extract
the name of a metric and its value. It assumes that each line follows
the same format (see vmstat or meminfo). It is very similar to how many
of use awk to extract such information.

This was written in response to the current vmstat collector
limiting the metrics to a very small subset. The idea is that this
can be used for any file, or can be extended to customize behaviour.
This could form the base class for many existing collectors. 

The two required config options are 'regex' (which has a default value),
and 'file' (with no default value).

### CODE REVIEW

File: src/collectors/genericfile/genericfile.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        return config_help
+
+    def get_default_config(self):
+        """
+        Returns the default collector settings
+        """
+        config = super(GenericFileCollector, self).get_default_config()
+        config.update({
+            'regex': '^\s*(?P<name>\w+)\s*[\s|:|=]\s*(?P<value>\w+)$',
+            'path': '',
```
Comment (shortdudey123) [inline src/collectors/genericfile/genericfile.py]: Not sure the top level is the best place for this

Comment (stantona) [inline src/collectors/genericfile/genericfile.py]: Yeah. I'm not sure what a good default path would be as the path will be dependent on the file. For example, if you're working with '/proc/meminfo', you could set the path to 'meminfo'.  I could give it some generic name, or remove it and treat it as a mandatory option that the user must set in the configuration.

Comment (shortdudey123) [inline src/collectors/genericfile/genericfile.py]: maybe `generic_file` for a default?

File: src/collectors/genericfile/genericfile.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        return config_help
+
+    def get_default_config(self):
+        """
+        Returns the default collector settings
+        """
+        config = super(GenericFileCollector, self).get_default_config()
+        config.update({
+            'regex': '^\s*(?P<key>\w+)\s*[\s|:|=]\s*(?P<value>\w+)\s*$',
+            'path': '',
```
Comment (shortdudey123) [inline src/collectors/genericfile/genericfile.py]: Any reason to yank the default of `generic_file` back out?

Comment (stantona) [inline src/collectors/genericfile/genericfile.py]: Ahh, my bad. I'll fix that.

Comment (shortdudey123) [inline src/collectors/genericfile/genericfile.py]: any update on this?

**Pull request closed without merging**